composer/utils/string_helpers.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Utilities for string manipulation."""
+
+
+def partial_format(s, *args, **kwargs):
+    """Format a string with a partial set of arguments.
+
+    Since `str.format()` raises a `KeyError` if a format key is missing from the arguments, this
+    function allows for a partial set of arguments to be provided. Any missing arguments will be
+    left as-is in the string.
+    """
+    result = s
+    done = False
+    while not done:
```

Comment:
I don't love the while loop here which in theory might be an infinite loop. is there any other way? maybe we just go back to hardcoded placeholder strings for mlflow instead of the proper {} string formatting?